### ISSUE TITLE
KLDivLoss input & target attention weights

### ISSUE THREAD

**Issue by hibanajjar998** [sarahwie/attention#6]
Hi Sarah, thanks for this valuable contribution!
I'm trying to reproduce the adversarial training experiment, and for the KL loss I saw that you switched the arguments from (input, target) as it is in [pytorch docs](https://pytorch.org/docs/stable/generated/torch.nn.KLDivLoss.html) to (target, input) in your code [here](https://github.com/sarahwie/attention/blob/d867a2b878d44169fbe4966d09f74cd036936bb7/model/Binary_Classification.py#L125). Since KL loss is not symmetric, the results will be different.
Therefore I wanted to ask whether you had a specific reason for which you swapped the two arguments? 
Thanks!

**Comment by sarahwie** [comment]
Hi, good catch and I believe this is a bug-- I don't think the pytorch documentation was as clear at the time that their implementation does not follow the standard mathematical notation, as the note on their docs page now indicates.

However, we also had discussions on whether the loss should contain the symmetric TVD term or just KLD, and we ultimately chose KLD as we hypothesized the choice wouldn't make much of a difference. I'd be curious if you find otherwise :)

**Comment by hibanajjar998** [comment]
Might indeed not make much difference. In my experiments what I'm comparing though for the adversarial training is using TVD or the original loss which is in my case MSE, since I have a regression task. Also swapped TVD with nn.L1loss at some point since it was more convenient for pytorch lightning logging.
I don't expect much differences but will report here if anything surprising comes out of it ^^
